glitter/publisher/models.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import unicode_literals
+
+from django.conf import settings
+from django.contrib.admin.models import CHANGE, LogEntry
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.utils.encoding import force_text, python_2_unicode_compatible
+
+from glitter.models import Version
+
+
+@python_2_unicode_compatible
+class PublishAction(models.Model):
+    UNPUBLISH_CHOICE = -1
+
+    # The glittered object we're linked to
+    content_type = models.ForeignKey(ContentType)
+    object_id = models.PositiveIntegerField()
+    content_object = GenericForeignKey('content_type', 'object_id')
+
+    # When/what we're updating, and by who
+    scheduled_time = models.DateTimeField()
+    publish_version = models.IntegerField()
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.PROTECT)
+
+    class Meta:
+        verbose_name = 'Scheduled publishing'
+        ordering = ('scheduled_time',)
+        unique_together = (
+            ('content_type', 'object_id', 'scheduled_time'),
+        )
+
+    def __str__(self):
+        if self.publish_version == self.UNPUBLISH_CHOICE:
+            return 'Unpublish at {}'.format(self.scheduled_time)
+        else:
+            return 'Publish version {} at {}'.format(self.publish_version, self.scheduled_time)
+
+    def save(self, *args, **kwargs):
+        super(PublishAction, self).save(*args, **kwargs)
+
+        # Add a task
+        if self.can_schedule_task():
```

Comment:
Not really.

The publish/process code is fairly defensive against multiple tasks. If the object changes date, then the queryset in `process_actions` won't see/select it. If the object is deleted as the task was already completed earlier - it won't be able to see/select it again.